packages/core/src/lib/jdk/JdkInstaller.ts

```
@@ -0,0 +1,96 @@
+ */
+
+'use strict';
+
+import * as path from 'path';
+import util = require('../util');
+
+const JDK_VERSION = '8u265-b01';
+const JDK_DIR = `jdk${JDK_VERSION}`;
+const DOWNLOAD_JDK_BIN_ROOT = `https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk${JDK_VERSION}/`;
+const DOWNLOAD_JDK_SRC_ROOT = 'https://github.com/AdoptOpenJDK/openjdk-jdk8u/archive/';
+const JDK_BIN_VERSION = JDK_VERSION.replace('-', '');
+const JDK_FILE_NAME_MAC = `OpenJDK8U-jdk_x64_mac_hotspot_${JDK_BIN_VERSION}.tar.gz`;
+const JDK_FILE_NAME_WIN32 = `OpenJDK8U-jdk_x86-32_windows_hotspot_${JDK_BIN_VERSION}.zip`;
+const JDK_FILE_NAME_LINUX64 = `OpenJDK8U-jdk_x64_linux_hotspot_${JDK_BIN_VERSION}.tar.gz`;
+const JDK_SRC_ZIP = `jdk${JDK_VERSION}.zip`;
+
+/**
+ * Checks whether JDK 8 is installed. If not installed,
+ * download the binary and source code and decompress at
+ * path given by user.
+ */
+export class JdkInstaller {
+  private process: NodeJS.Process;
+  private downloadFile: string;
+  private unzipFunction: (srcPath: string, dstPath: string, deleteWhenDone: boolean)
+    => Promise<void>;
+  private joinPath: (...paths: string[]) => string;
+  /**
+   * Constructs a new instance of JdkInstaller
+   *
+   * @param process {NodeJS.Process} process information from the OS process
+   */
+  constructor(process: NodeJS.Process) {
+    this.process = process;
+    this.unzipFunction = util.untar;
+    this.joinPath = path.posix.join;
+    switch (process.platform) {
+      case 'win32': {
+        this.downloadFile = JDK_FILE_NAME_WIN32;
+        this.unzipFunction = util.unzipFile;
+        this.joinPath = path.win32.join;
+        break;
+      }
+      case 'darwin': {
+        this.downloadFile = JDK_FILE_NAME_MAC;
+        break;
+      }
+      case 'linux': {
+        this.downloadFile = JDK_FILE_NAME_LINUX64;
+        break;
+      }
+      default:
+        this.downloadFile = '';
+        console.log('Platform not found. Cannot download appropriate JDK.');
```

Would it not be better to throw the error at this point instead of in `install`? We know that the `JdkInstaller` is going to be useless here, I don't see the point in letting it exist for a while in an invalid state.